docs/usage/validators.md

```diff
@@ -419,15 +292,120 @@ except ValidationError as e:
     """
 ```
 
-As with field validators, root validators can have `pre=True`, in which case they're called before field
-validation occurs (and are provided with the raw input data), or `pre=False` (the default), in which case
-they're called after field validation.
+Model validators can be `mode='before'` or `mode='after'` and `mode='wrap'`.
+
+Before model validators are passed the raw input which is often a `dict[str, Any]` but could also be an instance of the model itself (e.g. if `UserModel.model_validate(UserModel.construct(...))` is called) or anything else since you can pass arbitrary objects into `model_validate`.
+Because of this before validators are extremely flexible and powerful but can be cumbersome and error prone to implement.
+Before model validators should be class methods.
+The first argument should be `cls` (and we also recommend you use `@classmethod` below `@model_validator` for proper type checking), the second argument will be the input (you should generally type it as `Any` and use `isinstance` to narrow the type) and the third argument (if present) will be a `pydantic.ValidationInfo`.
+
+After validators are instance methods and always receive an instance of the model as the first argument.
+You should not use `(cls, ModelType)` as the signature, instead just use `(self)` and let type checkers infer the type of `self` for you.
+Since these are fully type safe they are often easier to implement than before validators.
```

Comment:
```suggestion
Since these are fully type safe they are often easier to implement than `mode='before'` model_validators.
```
It's easy for us to read the docs and understand when we are using the words "before", "after", "validator" as referencing specific concepts, but I think it might be a bit harder for people needing to go through these docs, so I would suggest trying to make it more clear when it's a reference to a specific concept (e.g., `mode='before'`, or `@model_validator`) rather than using "normal" language, since I think it will cost new readers some cognitive overhead to parse 
> After validators are instance methods ...
as
> `mode='after'` validators are instance methods ...
rather than
> After the point in time that validators are instance methods ...

Obviously they won't stay confused for long, I just think it's a pattern we can try to follow to make this all that much easier to read.